src/platform/plugins/shared/workflows_management/public/features/validate_workflow_yaml/model/types.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+export interface VariableItem {
+  id: string;
+  type: 'regexp' | 'foreach';
+  key: string | null;
+  startLineNumber: number;
+  startColumn: number;
+  endLineNumber: number;
+  endColumn: number;
+  yamlPath: (string | number)[];
+}
+
+export interface StepNameInfo {
+  name: string;
+  node: any;
+  startLineNumber: number;
+  startColumn: number;
+  endLineNumber: number;
+  endColumn: number;
+}
+
+export type YamlValidationErrorSeverity = 'error' | 'warning' | 'info';
+
+interface YamlValidationResultBase {
+  id: string;
+  startLineNumber: number;
+  startColumn: number;
+  endLineNumber: number;
+  endColumn: number;
+  hoverMessage: string | null;
+}
+
+interface YamlValidationResultNonUniqueStepName extends YamlValidationResultBase {
+  severity: 'error';
```

Comment:
NIT: Maybe it would make sense to define severity with `YamlValidationErrorSeverity` here, even if right now only `'error'` is possible, it simplifies the logic and avoids castings like this one:
https://github.com/elastic/kibana/pull/237134/files#diff-63ae21052a1e1a5bee3419ca081491a0fe9f73928485257a74da73301c41d63eR263